nixos/doc/rl-2211: cleanup (#203391)
NixOS/nixpkgs: nixos/doc/manual/release-notes/rl-2211.section.md

# Release 22.11 (“Raccoon”, 2022.11/??) {#sec-release-22.11}

This release is supported until the end of June 2023, handing over to NixOS 23.05.

## Highlights {#sec-release-22.11-highlights}

In addition to numerous new and upgraded packages, this release includes the following highlights:

- GNOME has been upgraded to 43. Please take a look at their [Release
  Notes](https://release.gnome.org/43/) for details.

- KDE Plasma has been upgraded to 5.25 and 5.26.
    - https://kde.org/announcements/plasma/5/5.25.0/

>     - https://kde.org/announcements/plasma/5/5.26.0/

Reviewer: We should probably only mention the latest version here. cc @K900

Reviewer 2: The changelogs aren't cumulative, so you'll have to read both if you want to figure out what changed.

Reviewer 2: Yes, we can say "upgraded from 5.24 to 5.26" and then link both changelogs.

- Cinnamon has been updated to 5.4 and the Cinnamon module now defaults to
  blueman as Bluetooth manager and slick-greeter as LightDM greeter to match upstream.

- OpenSSL now defaults to OpenSSL 3, updated from 1.1.1.

- PHP now defaults to PHP 8.1, updated from 8.0.

- PHP is now built `NTS` (Non-Thread Safe) style by default, for Apache and
  `mod_php` usage we still enable `ZTS` (Zend Thread Safe). This has been a
  common practice for a long time in other distributions.

> - PHP 8.2.0 RC 7 is available.

Reviewer: Not sure whether we should mention release candidate availability.

Author: Removed.

> - PHP 8.2.0 RC 7 is available.

Reviewer 3: I think we just need one paragraph for php. Also agree that we can drop the release candidate.

Author: I think the version bump and thread mode stuff are fine separate. Don't think it matters too much, though. Thoughts?

> - `protonup` has been aliased to and replaced by `protonup-ng` due to upstream not maintaining it.

Reviewer 3: Never heard of this package before. Is this a notable change?

Reviewer 4: It's not something that is likely to affect you but it should be noted that this happened.

- Perl has been updated to 5.36, and its core module `HTTP::Tiny` was patched to verify SSL/TLS certificates by default.

- `emacs` enables native compilation which means:
  - Emacs packages from Nixpkgs, builtin or not, will do native compilation ahead of time so you can enjoy the benefit of native compilation without compiling them on you machine;
  - Emacs packages from somewhere else, e.g. `package-install`, will do asynchronously deferred native compilation. If you do not want this, maybe to avoid CPU consumption for compilation, you can use `(setq native-comp-deferred-compilation nil)` to disable it while still enjoy the benefit of native compilation for packages from Nixpkgs.

- `nixos-generate-config` now generates configurations that can be built in pure

>   mode. This is achieved by setting the new `nixpkgs.hostPlatform` option.

Reviewer 3: I think the implication of what pure mode means should be explained here.

Author: Do we have any documentation for this that isn't some GitHub issue, though? Don't think the flakes manual has anything on it...

  You may have to unset the `system` parameter in `lib.nixosSystem`, or similarly

>   remove definitions of the `nixpkgs.{system,localSystem,crossSystem}` options.

Reviewer 3: This is for nixos systems that you want to cross-compile?

Author: No clue, will look into it.

  Alternatively, you can remove the `hostPlatform` line and use NixOS like you
  would in NixOS 22.05 and earlier.

> - An image configuration and generator has been added for Linode images, largely based on the present GCE configuration and image.

Reviewer 3: ```suggestion
- It is now possible to generate NixOS images targeting the Linode cloud provider. This is image is largely based on the present GCE configuration and image.
```

Author: I reworded it differently, but thanks for bringing my attention to it.

# Release 22.11 (“Raccoon”, 2022.11/??) {#sec-release-22.11}

> The NixOS release team is happy to announce a new version of NixOS 22.11. NixOS is both a linux distribution, and a set of packages usable on other Linux systems and macOS.

Author: I'm not sure if we want this, if only because we have similar stuff in the blog post.

Reviewer: I'm not opposed.

    - This release introduces warnings for the use of deprecated hash algorithms for both methods of configuring passwords. To make sure you migrated correctly, run `nixos-rebuild switch`.

- The NixOS documentation is now generated from markdown. While docbook is still part of the documentation build process, it's a big step towards the full migration.

- `aarch64-linux` is now included in the `nixos-22.11` and `nixos-22.11-small` channels. This means that when those channel update, both `x86_64-linux` and `aarch64-linux` will be available in the binary cache.

- `aarch64-linux` ISOs are now available on the [downloads page](https://nixos.org/download.html).

- `nsncd` is now available as a replacement of `nscd`.

  `nscd` is responsible for resolving hostnames, users and more in NixOS and has been a long standing source of bugs, such as sporadic network freezes.
  
  More context in this [issue](https://github.com/NixOS/nixpkgs/issues/135888).
  
  Help us test the new implementation by setting `services.nscd.enableNsncd` to `true`.

  We plan to use `nsncd` by default in NixOS 23.05.

> - Linode cloud images are now supported by importing `${modulesPath}/virtualisation/linode-image.nix` and accessing `system.build.linodeImage` on the output.

Author: These instructions aren't clear enough, I fear.

When I was going over this section, I concluded that there's no succulent way to provide instructions to build an image...

Reviewer 3: `nixos-generate -o linode` would be an easier example: https://github.com/nix-community/nixos-generators

  for a transition period so that in time the ecosystem can switch without
  breaking compatibility with any supported NixOS release.

## Notable version updates {#sec-release-22.11-version-updates}

> - Nix has been upgraded from [v2.8.1 to v2.11.0](https://github.com/NixOS/nix/compare/2.8.1...2.11.0)

Author: We should probably use the same/similar wording for this pattern throughout:

```suggestion
- Nix has been upgraded from v2.8.1 to v2.11.0. For more information, please see [the release notes](https://github.com/NixOS/nix/compare/2.8.1...2.11.0).
```

- OpenSSL now defaults to OpenSSL 3, updated from 1.1.1.

> - GNOME has been upgraded to version 43. Please take a look at their [Release Notes](https://release.gnome.org/43/) for details.

Author: ```suggestion
- GNOME has been upgraded to version 43. Please see the [release notes](https://release.gnome.org/43/) for more information.
```

- Perl has been updated to 5.36, and its core module `HTTP::Tiny` was patched to verify SSL/TLS certificates by default.

- Improved performances of `lib.closePropagation` which was previously quadratic. This is used in e.g. `ghcWithPackages`. Please see backward incompatibilities notes below.

- Cinnamon has been updated to 5.4. While at it, the cinnamon module now defaults to
  blueman as bluetooth manager and slick-greeter as lightdm greeter to match upstream.

- OpenSSL now defaults to OpenSSL 3, updated from 1.1.1.

- An image configuration and generator has been added for Linode images, largely based on the present GCE configuration and image.

- `hardware.nvidia` has a new option `open` that can be used to opt in the opensource version of NVIDIA kernel driver. Note that the driver's support for GeForce and Workstation GPUs is still alpha quality, see [NVIDIA Releases Open-Source GPU Kernel Modules](https://developer.nvidia.com/blog/nvidia-releases-open-source-gpu-kernel-modules/) for the official announcement.

<!-- To avoid merge conflicts, consider adding your item at an arbitrary place in the list instead. -->

> ## New Services {#sec-release-22.11-new-services}

Reviewer 3: Would it make sense to just mention how many services has been added in the last release and than maybe mention a couple interesting ones?

Author: Moved to the end, not sure if we want to do this given time constraints. cc @mweinelt

- The `fetchgit` fetcher now uses [cone mode](https://www.git-scm.com/docs/git-sparse-checkout/2.37.0#_internalscone_mode_handling) by default for sparse checkouts. [Non-cone mode](https://www.git-scm.com/docs/git-sparse-checkout/2.37.0#_internalsnon_cone_problems) can be enabled by passing `nonConeMode = true`, but note that non-cone mode is deprecated and this option may be removed alongside a future Git update without notice.

> - The `fetchgit` fetcher supports sparse checkouts via the `sparseCheckout` option. This used to accept a multi-line string with directories/patterns to check out, but now requires a list of strings

Author: Why remove the period? We (hopefully) are using periods at the end of every entry.

> - `openssh` was updated to version 9.1, disabling the generation of DSA keys when using `ssh-keygen -A` as they are insecure. Also, `SetEnv` directives in `ssh_config` and `sshd_config` are now first-match-wins

Author: Speaking of periods, this doesn't (seem to) have one. Do you mind making sure every entry does? (I didn't think many were missing them when I looked, FWIW.)
